src/averageAges.js

```diff
@@ -56,6 +83,26 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const mothers = people.reduce((acc, person) => {
+    acc[person.name] = person;
+
+    return acc;
+  }, {});
+
+  const motherWithChildren = people.filter(child =>
+    onlyWithSon
+      ? mothers[child.mother]
+         && child.sex === 'm'
+      : mothers[child.mother]
+  );
+
+  const calculateAge = motherWithChildren.map(person =>
+    person.born - people.find(mother => mother.name === person.mother).born);
+
+  const averageAge = calculateAge.reduce((sum, age) =>
+    sum + age) / calculateAge.length;
+
+  return averageAge;
```

Comment:
Instead of three loops
```suggestion
  let sum = 0;

  const children = people.filter(p => {
    const isPerson = onlyWithSon
      ? p.sex === 'm' && mothers[p.mother]
      : mothers[p.mother];

    if (isPerson) {
      sum += p.born - mothers[p.mother].born;

      return true;
    }

    return false;
  });

  return sum / children.length;
```

Reply:
You asked about example. Instead of three loops it can be ome loop.
Also instead filter you can use reduce.
Anyway good job, I just wanted to think about performance in future)